lib/matplotlib/axes/_axes.py

```diff
@@ -5843,6 +5846,8 @@ def hist(self, x, bins=10, range=None, normed=False, weights=None,
                     xvals.append(x.copy())
                     yvals.append(y.copy())
 
+            fill_kwargs = dict() if not input_empty else dict(linewidth=0)
```

Comment:
As above, it draws one patch with linewidth 1 if it is not manually set to 0 and thus is not an empty graph.

Before (linewidth 1, non empty graph):
![step_linewidth_nonzero](https://cloud.githubusercontent.com/assets/5559279/6648379/57d55596-c9af-11e4-8909-2917e17f1955.png)

After (linewidth 0, empty graph):
![step_linewidth_zero](https://cloud.githubusercontent.com/assets/5559279/6648380/57d9d8aa-c9af-11e4-94c9-a0b58b2f2adf.png)